internal/injector/aspect/advice/call.go

```diff
@@ -40,13 +45,65 @@ func (a *appendArgs) Apply(ctx context.Context, chain *node.Chain, csor *dstutil
 		}
 	}
 
-	call.Args = append(call.Args, newArgs...)
+	if !call.Ellipsis {
+		call.Args = append(call.Args, newArgs...)
+		return true, nil
+	}
+
+	// The function call has an ellipsis, so we need to append our new arguments to the last argument,
+	// which is a slice. To do so, we need to provision a new slice of the right type and size, append
+	// all the relevant data in there, and then replace the last argument with the new slice.
+	lastIdx := len(call.Args) - 1
+	call.Args[lastIdx] = &dst.CallExpr{
```

Comment:
So IIUC, this block is trying to replace something like `opts...` by `append(opts, newArgs)...`, in which case I suggest we enforce copying `opts` instead with the same trick you used in the code below by replacing by `append(make([]<type of opts>, 0, len(opts)+1), opts, newArgs)...`

Reply:
This is wrapped in an IIFE which receives the original options variadically, implying a slice copy already.